src/features/student/declaredPrograms/components/interactions/formFields/DeclaredProgramLinkFormField/DeclaredProgramLinkFormField.test.ts

```diff
@@ -0,0 +1,153 @@
+import type { AddDeclaredProgramForm, DeclaredProgramFormData } from '@/features/student/declaredPrograms/types/forms.types'
+import DeclaredProgramLinkFormField from '@/features/student/declaredPrograms/components/interactions/formFields/DeclaredProgramLinkFormField/DeclaredProgramLinkFormField.vue'
+import { BddTest } from '@avenirs-esr/avenirs-dsav/test-utils'
+import { useForm } from '@tanstack/vue-form'
+import { mount, type VueWrapper } from '@vue/test-utils'
+import { beforeEach, expect, vi } from 'vitest'
+
+const DeclaredProgramLinkInputStub = defineComponent({
+  name: 'DeclaredProgramLinkInput',
+  props: {
+    modelValue: String,
+    errorMessage: String
+  },
+  emits: ['update:modelValue', 'blur'],
+  template: `
+    <div data-testid="declared-program-link-input-stub">
+      <input
+        :value="modelValue"
+        @input="$emit('update:modelValue', $event.target.value)"
+        @blur="$emit('blur')"
+      />
+    </div>
+  `
+})
+
+const TestWrapper = defineComponent({
+  components: {
+    DeclaredProgramLinkFormField
+  },
+  setup () {
+    const form = useForm({
+      defaultValues: {
+        link: ''
+      } as DeclaredProgramFormData,
+      validators: {
+        onSubmit () {
+          return { fields: { link: undefined } }
+        }
+      }
+    }) as unknown as AddDeclaredProgramForm
+
+    return { form }
+  },
+  template: `
+    <form @submit.prevent="form.handleSubmit">
+      <DeclaredProgramLinkFormField :form="form" />
+    </form>
+  `
+})
+
+BddTest().given('a declared program link form field', () => {
+  let wrapper: VueWrapper<InstanceType<typeof TestWrapper>>
+
+  const stubs = {
+    DeclaredProgramLinkInput: DeclaredProgramLinkInputStub
+  }
+
+  BddTest().when('the component is mounted', () => {
+    beforeEach(() => {
+      vi.clearAllMocks()
+      wrapper = mount(TestWrapper, {
+        global: { stubs }
+      })
+    })
+
+    BddTest().then('it should render the link input component', () => {
+      const input = wrapper.findComponent({ name: 'DeclaredProgramLinkInput' })
+      expect(input.exists()).toBe(true)
+    })
+
+    BddTest().then('it should have empty initial value', () => {
+      const input = wrapper.findComponent({ name: 'DeclaredProgramLinkInput' })
+      expect(input.props('modelValue')).toBe('')
+    })
+  })
+
+  BddTest().and('the user enters a link', () => {
+    beforeEach(async () => {
+      vi.clearAllMocks()
+      wrapper = mount(TestWrapper, {
+        global: { stubs }
+      })
```

Comment:
```suggestion

  BddTest().and('the user enters a link', () => {
```

idem que les autres fichiers de test : and dans when et suppression du code répété. Valable pour tous les and